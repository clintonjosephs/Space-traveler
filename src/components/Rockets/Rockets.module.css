.listContainer {
  margin: 0.6rem 6.2rem;
}

.img {
  margin-right: 1rem !important;
  border-top-left-radius: 5px !important;
  border-bottom-left-radius: 5px !important;
}

.listItems {
  display: flex;
  flex-direction: row;
  margin-bottom: 2rem;
  box-shadow: 0 0 10px var(--neutral-color-2);
  border-radius: 5px;
  transition: all 0.2s ease-in-out;
}

.listItems:hover {
  transform: scale(1.05);
}

.listWrapper {
  padding: 0.8rem;
  line-height: 1.5rem;
}

.listWrapper div {
  font-size: 1.2rem;
  padding: 1rem 0 1rem;
}

.listWrapper button {
  width: 12rem;
  height: 3rem;
  box-shadow: none;
  font-size: 1rem;
  border-radius: 5px;
  cursor: pointer;
  margin: 0.2rem;
}

.reserve {
  background: var(--accent-color);
  color: var(--white);
  border: none;
}

.cancel {
  color: var(--neutral-color-2);
  background: var(--white);
  border: solid 3px var(--neutral-color);
}

.badge {
  background: var(--accent-color-2);
  color: var(--white);
  border-radius: 5px;
  padding: 5px;
  font-size: 0.8rem;
  margin-right: 0.3rem;
}

.describe {
  line-height: 2rem;
  text-align: justify;
}

@media (max-width: 786px) {
  .listContainer {
    margin: 0;
    padding: 0;
    width: 100%;
  }

  .img {
    height: 100%; 
    overflow: auto;
    width: 100% !important;
    margin-right: 1rem !important;
  }

  .listItems {
    display: flex;
    flex-direction: column;
  }
}
