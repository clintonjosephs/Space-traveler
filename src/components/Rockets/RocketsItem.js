import React from 'react';
import PropTypes from 'prop-types';
import styles from './Rockets.module.css';

const RocketsItem = (props) => {
  const {
    rocketId, rocketName, description, flickerImages,
  } = props;
  const clickHandler = () => {
    console.log(rocketId);
  };
  return (
    <li className={styles.listItems}>
      <img src={flickerImages} alt={rocketName} />
      <div className={styles.listWrapper}>
        <h1>{rocketName}</h1>
        <div>
          <span className={styles.badge}>Reserved</span>
          {description}
        </div>
        <button type="button" className={styles.reserve} onClick={clickHandler}>
          Reserve Rocket
        </button>
        <button type="button" className={styles.cancel} onClick={clickHandler}>
          Cancel Reservation
        </button>
      </div>
    </li>
  );
};

RocketsItem.propTypes = {
  rocketId: PropTypes.string.isRequired,
  rocketName: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  flickerImages: PropTypes.string.isRequired,
};

export default RocketsItem;
