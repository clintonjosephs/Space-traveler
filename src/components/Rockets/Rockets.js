import React from 'react';
import { useSelector } from 'react-redux';
import RocketsItem from './RocketsItem';
import styles from './Rockets.module.css';

const Rockets = () => {
  const data = useSelector((state) => state.rocketReducer.rockets);

  if (data.length === 0) {
    return (
      <>
        <ul className={styles.listContainer}>
          <li className={styles.listItems} style={{ padding: '2rem' }}><h1>No rockets available</h1></li>
        </ul>
      </>
    );
  }
  return (
    <>
      <ul className={styles.listContainer}>
        {data.map(({
          id, rocketName, description, flickrImages,
        }) => (
          <RocketsItem
            key={id}
            rocketName={rocketName}
            description={description}
            flickerImages={flickrImages}
            id={id}
          />
        ))}
      </ul>
    </>
  );
};

export default Rockets;
