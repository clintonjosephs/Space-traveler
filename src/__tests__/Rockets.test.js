import React from 'react';
import { Provider } from 'react-redux';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import store from '../redux/ConfigureStore';
import Rockets from '../components/Rockets/Rockets';

const ReduxProvider = () => (
  <Provider store={store}>
    <Rockets />
  </Provider>
);

describe('Check rocket snapshot', () => {
  it('matches snapshot', () => {
    const component = renderer
      .create(
        <ReduxProvider />,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});

describe('Rockets Component', () => {
  it('renders rockets', async () => {
    render(<ReduxProvider />);
    const allRockets = await screen.findAllByText('Reserve Rocket');
    expect(allRockets).toHaveLength(4);
  });

  test('user operation on click rocket reservation and cancel reservation', async () => {
    render(<ReduxProvider />);

    const reserveButton = await screen.findAllByText(/Reserve Rocket/i);
    fireEvent.click(reserveButton[0]);
    const badges = await screen.findAllByRole('switch');
    expect(badges).toHaveLength(1);

    expect(reserveButton[0].textContent).toBe('Cancel Reservation');
    fireEvent.click(reserveButton[0]);

    expect(badges[0].style.display).toBe('none');
    expect(reserveButton[0].textContent).toBe('Reserve Rocket');
  });
});
