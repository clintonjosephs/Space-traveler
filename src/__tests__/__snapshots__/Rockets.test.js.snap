// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check rocket snapshot matches snapshot 1`] = `
<ul
  className="listContainer"
>
  <li
    className="listItems"
    style={
      Object {
        "padding": "2rem",
      }
    }
  >
    <h1>
      No rockets available
    </h1>
  </li>
</ul>
`;
